### PR TITLE
🎉 Source Shopify: add missing fields to the `Products` stream

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: 
- https://github.com/airbytehq/airbyte-internal-issues/issues/7738

## How
- extended the `products` query to include the missing fields
- extended the `products` stream schema to include the missing fields
- added processing for them (unnest is required for some of them)
- edited `unit_test` case to cover the changes

## User Impact
No impact is expected, but the `reset` is recommended in order to pick up the schema changes and have the target fields.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚

### CODE REVIEW

**Review by ChristoGrab — APPROVED** [review]
Verified that the fields are all included in the response, one small nit but otherwise all LGTM!

File: airbyte-integrations/connectors/source-shopify/pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ requires = [ "poetry-core>=1.0.0",]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
-version = "2.1.0"
+version = "2.1.2"
```
Comment (ChristoGrab) [inline airbyte-integrations/connectors/source-shopify/pyproject.toml:6]: Teeny nit; should this be set to 2.1.1?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/pyproject.toml:6]: Nope, this is the next one after this one: https://github.com/airbytehq/airbyte/pull/38352

**Review by cmm-airbyte — COMMENTED** [review]
LGTM

**Review by maxi297 — APPROVED** [review]
LGTM!

**Review by aldogonzalez8 — APPROVED** [review]

**Pull request merged**